File: game-server/src/main/java/org/forfun/mmorpg/game/ai/fsm/Atttack2PatrolTransition.java
package org.forfun.mmorpg.game.ai.fsm;

import org.forfun.mmorpg.game.database.user.entity.PlayerEnt;
import org.forfun.mmorpg.game.scene.actor.Creature;
import org.forfun.mmorpg.game.scene.actor.Monster;

public class Atttack2PatrolTransition extends Transition {

	public Atttack2PatrolTransition(State from, State to) {
		super(from, to);
	}

	@Override
	public boolean meetCondition(Creature creature) {
		PlayerEnt player = (PlayerEnt) creature;
		Scene scene = player.getScene();
		Monster monster = scene.getMonster();

		return monster.isDie();
	}

}

File: game-server/src/main/kotlin/org/forfun/mmorpg/game/ai/fsm/Atttack2PatrolTransition.kt
package org.forfun.mmorpg.game.ai.fsm

import org.forfun.mmmorpg.game.scene.actor.Creature
import org.forfun.mmorpg.game.database.user.entity.PlayerEnt

class Atttack2PatrolTransition(from: State, to: State) : Transition(from, to) {

    override fun meetCondition(creature: Creature): Boolean {
        val player = creature as PlayerEnt
        val scene = player.scene
        val monster = scene?.monster

        return monster!!.isDie
    }
}
